packages/multichain-network-controller/src/MultichainNetworkService/MultichainNetworkService.test.ts

```diff
@@ -59,6 +81,92 @@ describe('MultichainNetworkService', () => {
       expect(result).toStrictEqual(mockResponse);
     });
 
+    it('batches requests when account IDs exceed the default batch size', async () => {
+      const manyAccountIds: CaipAccountId[] = [];
+      for (let i = 1; i <= 30; i++) {
+        manyAccountIds.push(
+          `${KnownCaipNamespace.Eip155}:${i}:${MOCK_EVM_ADDRESS}` as CaipAccountId,
+        );
+      }
+
+      const batches = chunk(manyAccountIds, DEFAULT_BATCH_SIZE);
+
+      const firstBatchResponse = {
+        activeNetworks: batches[0],
+      };
+      const secondBatchResponse = {
+        activeNetworks: batches[1],
+      };
+
+      mockFetch
+        .mockResolvedValueOnce({
+          ok: true,
+          json: () => Promise.resolve(firstBatchResponse),
+        })
+        .mockResolvedValue({
+          ok: true,
+          json: () => Promise.resolve(secondBatchResponse),
+        });
+
+      const service = new MultichainNetworkService({
+        fetch: mockFetch,
+      });
+
+      const result = await service.fetchNetworkActivity(manyAccountIds);
+
+      expect(mockFetch).toHaveBeenCalled();
```

Comment:
Nit: Do we want to check it has been called 2 times?

```suggestion
      expect(mockFetch).toHaveBeenCalledTimes(2);
```

Reply:
Yes that would be much better. I've added here [6be2f4e](https://github.com/MetaMask/core/pull/5752/commits/6be2f4ed8519802090bbfa4cd5c139766ce3cc9b)